addOns/bruteforce/src/main/java/org/zaproxy/zap/extension/bruteforce/ExtensionBruteForce.java

```diff
@@ -152,16 +168,158 @@ public void optionsLoaded() {
         }
     }
 
-    protected void bruteForceSite(SiteNode siteNode) {
-        this.getBruteForcePanel().bruteForceSite(siteNode);
+    public int bruteForceSite(SiteNode siteNode, String payloadFileName) {
+        for (ForcedBrowseFile file : this.getFileList()) {
+            if (file.getFile().getName().equals(payloadFileName)) {
+                return this.bruteForceSite(siteNode, file.getFile());
+            }
+        }
+        throw new IllegalArgumentException("File not found: " + payloadFileName);
+    }
+
+    public int bruteForceSite(SiteNode siteNode, File payloadFile) {
+        ScanTarget scanTarget = createScanTarget(siteNode);
+        BruteForce bf = this.startScan(scanTarget, null, payloadFile, false);
+        return bf.getScanId();
+    }
+
+    private BruteForce getBruteForce(int scanId) {
+        BruteForce bf = this.bruteForceIndexes.get(scanId);
+        if (bf != null) {
+            return bf;
+        }
+        throw new IllegalArgumentException("ScanId not found: " + scanId);
+    }
+
+    public boolean isRunning(int scanId) {
+        return !this.getBruteForce(scanId).isStopped();
+    }
+
+    public boolean isPaused(int scanId) {
+        return this.getBruteForce(scanId).isPaused();
+    }
+
+    public int getProgress(int scanId) {
+        BruteForce bf = this.getBruteForce(scanId);
+        return 100 * bf.getWorkDone() / bf.getWorkTotal();
+    }
+
+    public boolean stopScan(int scanId) {
+        BruteForce bf = this.getBruteForce(scanId);
+        if (bf.isStopped()) {
+            return false;
+        }
+        bf.stopScan();
+        return true;
+    }
+
+    public boolean pauseScan(int scanId) {
+        BruteForce bf = this.getBruteForce(scanId);
+        if (bf.isPaused()) {
+            return false;
+        }
+        bf.pauseScan();
+        return true;
+    }
+
+    public boolean resumeScan(int scanId) {
+        BruteForce bf = this.getBruteForce(scanId);
+        if (!bf.isPaused()) {
+            return false;
+        }
+        bf.unpauseScan();
+        return true;
     }
 
-    protected void bruteForceDirectory(SiteNode siteNode) {
-        this.getBruteForcePanel().bruteForceDirectory(siteNode);
+    public List<ScanTarget> getActiveScans() {
+        return this.activeScans;
     }
 
-    protected void bruteForceDirectoryAndChildren(SiteNode siteNode) {
-        this.getBruteForcePanel().bruteForceDirectoryAndChildren(siteNode);
+    public BruteForce getBruteForce(ScanTarget target) {
```

Comment:
This (and others using `ScanTarget`) could be just package visible to not expose `ScanTarget`, the methods with the ID of the scan should be enough.